File: mobile/lib/features/quiz/domain/models/quiz_question.dart
final class QuizQuestion {
  QuizQuestion({
    required this.id,
    required this.prompt,
    required List<QuizOption> options,
  }) : options = List.unmodifiable(options);

  final String id;
  final String prompt;
  final List<QuizOption> options;
}

final class QuizOption {
  const QuizOption({required this.id, required this.text});

  final String id;
  final String text;
}

File: backend/src/AsliApp.Domain/Education/QuizQuestion.cs
namespace AsliApp.Domain.Education;

public sealed class QuizQuestion
{
    public Guid Id { get; set; }
    public Guid QuizId { get; set; }
    public string Prompt { get; set; } = string.Empty;
    public int Order { get; set; }
    public bool IsDeleted { get; set; }
    public DateTimeOffset? DeletedAtUtc { get; set; }
    public DateTimeOffset CreatedAtUtc { get; set; }
    public DateTimeOffset UpdatedAtUtc { get; set; }

    public Quiz Quiz { get; set; } = null!;
    public ICollection<QuizOption> Options { get; set; } = [];
}
